client/src/App.vue

```diff
@@ -22,19 +23,54 @@ export default {
     }
   },
   methods: {
-    loadData: function () {
-      console.log('loadData triggered:  this.$route.meta ', this.$route.meta)
+    loadData: async function () {
       const debugApp = false
       let params2 = getIncomingParams()
       StoreHelper.setLoading(null, true)
+      // API return to url
+      const apiUrl = params2['apiUrl'] || sessionStorage.getItem(sKeys.API_URL)
+      const refreshToken = params2['token']
+      let visitId = ''
+      const authToken = StoreHelper.getAuthToken()
       return Promise.resolve()
         .then(() => {
-          // API return to url
-          let apiUrl = params2['apiUrl']
+          if (!(apiUrl || sessionStorage.getItem(sKeys.API_URL))) {
+            return Promise.reject('Please provide the API\'s url')
+          }
+        })
+        .then(() => {
+          if (refreshToken) {
+            return StoreHelper.fetchToken(refreshToken, apiUrl)
+              .then(() => {
+                const token = StoreHelper.getAuthToken()
+                if (!token) {
+                  return Promise.reject('Refresh token is expired')
+                } else {
+                  return StoreHelper.fetchTokenData(token, apiUrl)
+                }
+              })
+          } else if (authToken) {
+            setAuthHeader()
+            return StoreHelper.fetchTokenData(authToken, apiUrl)
+          } else {
+            return Promise.reject('Parameters Error')
+          }
+        })
+        .then(() => {
+          return StoreHelper.getAuthPayload()
+        })
+        .then((payload) => {
+          if (!payload) {
+            return Promise.reject('Error when fetching token data')
+          } else {
+            visitId = payload.visitId
+          }
+        })
+        .then(() => {
           return this._loadApiUrl(apiUrl)
         })
         .then(() => {
-          let visitId = params2['visit']
+          // let visitId = params2['visit'] || visitId
```

Comment:
Revisit the following and see if it is needed. The visit id exists as of line 66 or else user doesn't get here. So remove the further checks